cmd/register.go

```diff
@@ -0,0 +1,132 @@
+			} else if quiet {
+				logLevel = log.FatalLevel
+			}
+			log.SetLevel(logLevel)
+
+			sess := session.Must(session.NewSessionWithOptions(session.Options{
+				SharedConfigState: session.SharedConfigEnable,
+			}))
+			svc := ecs.New(sess)
+			return register.run(args, svc)
+		},
+	}
+
+	f := cmd.Flags()
+	f.BoolVar(&register.strict, "strict", false, "fail on lint warnings")
+	f.StringSliceVarP(&register.balanceFiles, "balances", "f", []string{}, "specify values in a JSON file or an S3 URL")
+	f.StringSliceVar(&register.values, "set", []string{}, "set values on the command line (can repeat or use comma-separated values)")
+	f.StringSliceVar(&register.stringValues, "set-string", []string{}, "set STRING values on the command line (can repeat or use comma-separated values)")
+	f.BoolVar(&register.dryRun, "dry-run", false, "Do not actually register task definition; just print resulting task definition")
+	return cmd
+}
+
+func (r *registerCmd) registerTaskDefinition(taskDefnJSON string, svc ecsiface.ECSAPI) (string, error) {
+	var balances map[string]interface{}
+	balances, err := mergeValues(r.balanceFiles, r.values, r.stringValues)
+	if err != nil {
+		return "", err
+	}
+	values := map[string]interface{}{
+		"Values": balances,
+	}
+	log.Debugf("Values used for template: %#v", values)
+
+	registerTaskDefinitionInput, err := tasks.ParseTaskDefinition(taskDefnJSON, values, r.strict)
+	if err != nil {
+		return "", errors.Wrap(err, "cannot parse task definition")
+	}
+
+	if r.dryRun {
+		fmt.Printf("%+v\n", registerTaskDefinitionInput)
+		return "", nil
+	}
+
+	log.Debugf("Task definition: %+v", registerTaskDefinitionInput)
+	registerTaskDefinitionOutput, err := svc.RegisterTaskDefinition(registerTaskDefinitionInput)
+	if err != nil {
+		return "", errors.Wrap(err, "cannot register task definition")
+	}
+	taskDefn := registerTaskDefinitionOutput.TaskDefinition
+	log.Infof("Successfully registered task definition %#v", *taskDefn.TaskDefinitionArn)
+	return *taskDefn.TaskDefinitionArn, nil
+}
+
+func mergeValues(balanceFiles []string, values []string, stringValues []string) (map[string]interface{}, error) {
+	base := map[string]interface{}{}
+	for _, filePath := range balanceFiles {
+		balances, err := tasks.ParseBalances(filePath)
+		if err != nil {
+			return nil, err
+		}
+		if err := mergo.Merge(&base, balances, mergo.WithOverride); err != nil {
+			return nil, err
+		}
+	}
+	for _, value := range values {
+		if err := strvals.ParseInto(value, base); err != nil {
+			return nil, errors.Wrap(err, "failed parsing --set data")
+		}
+	}
+	for _, value := range stringValues {
+		if err := strvals.ParseIntoString(value, base); err != nil {
+			return nil, errors.Wrap(err, "failed parsing --set-string data")
+		}
+	}
+	return base, nil
+}
+
+func (r *registerCmd) run(args []string, svc ecsiface.ECSAPI) error {
+	taskDefnArn, err := r.registerTaskDefinition(args[0], svc)
+	if err != nil {
+		return err
+	}
+	if !r.dryRun {
```

Comment:
not sure I get what dryRun does here, haven't all the computations already run at this point?

Reply:
It will not actually register the task definition, instead just printing the rendered task definition.